fontbe/src/mvar.rs

```diff
@@ -0,0 +1,329 @@
+//! Generates an [MVAR](https://learn.microsoft.com/en-us/typography/opentype/spec/MVAR) table.
+
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+
+use fontdrasil::orchestration::AccessBuilder;
+
+use fontdrasil::{
+    coords::NormalizedLocation,
+    orchestration::{Access, Work},
+    types::Axis,
+};
+use fontir::{
+    ir::GlobalMetricValues, orchestration::WorkId as FeWorkId, variations::VariationModel,
+};
+use write_fonts::types::MajorMinor;
+use write_fonts::{
+    tables::{
+        mvar::{Mvar, ValueRecord},
+        variations::{ivs_builder::VariationStoreBuilder, VariationRegion},
+    },
+    types::Tag,
+    OtRound,
+};
+
+use crate::{
+    error::Error,
+    orchestration::{AnyWorkId, BeWork, Context, WorkId},
+};
+
+#[derive(Debug)]
+struct MvarWork {}
+
+pub fn create_mvar_work() -> Box<BeWork> {
+    Box::new(MvarWork {})
+}
+
+/// Helper to build MVAR table from global metrics sources.
+struct MvarBuilder {
+    /// Variation axes
+    axes: Vec<Axis>,
+    /// Sparse variation models, keyed by the set of locations they define
+    models: HashMap<BTreeSet<NormalizedLocation>, VariationModel>,
```

Comment:
I assume it's a hashmap (unstable order) of btreesets (stable order) for a good reason but the comment doesn't tell me what that reason is, perhaps it could?

Reply:
well, because it must be hashable otherwise it can't be used as a key in a hashmap? we do similar things all over fontc